draft/0001-organization.md

```diff
@@ -0,0 +1,273 @@
+
+## 4. Organization
+
+### 4.1. Definition
+
+An organization is an entity that can connect multiple identities. Each connection is called a membership, and the identity is called an organization member.
+
+### 4.2. Organization template
+
+In modern SaaS applications, it's common to create multiple organizations with the same set of permissions and roles (the template). For instance, a user may have access to multiple workspaces in Slack, each with varying levels of access. In such cases, roles and permissions must be defined within the context of an organization.
+
+While duplicating the template for each organization is feasible, it can become challenging to update the template consistently. For example, adding a new role called "collaborator" would require repeating the same action for every organization. Maintaining consistency across all organizations is crucial.
+
+An organization template is not associated with any specific organization; it is a resource at the tenant level that comprises organization permissions and roles.
+
+> This specification focuses on a single organization template scenario. Multiple organization templates within a tenant are beyond the scope of this discussion.
+
+### 4.3. Organization permission
+
+An organization permission SHOULD be represented as a meaningful string, also serving as the name and unique identifier within the organization template. Permission names should be easily understandable. For example, `read:books` indicates the action to read or retrieve books in the system, while `update:books` signifies updating book information.
+
+It is RECOMMENDED to only use lowercase letters with specific symbols (`:_`) for permissions.
+
+### 4.4. Organization role
+
+An organization role has a unique identifier and contains a set of organization permissions. Each organization role in the template SHOULD only be granted with predefined permissions in the same template. An organization role MAY have no permission granted.
+
+### 4.5. Assign roles to organization members
+
+Before assigning roles to an identity, the identity MUST have a membership with the organization, making them an organization member. An organization member can be assigned multiple roles within the organization, and each role held by the member must be predefined in the organization template. The complete set of permissions granted to an organization member is the union of all permissions granted by their roles.
+
+### 4.6. Role-based access control
+
+To authorize an actor (identity) to perform a specific action within an organization context, the actor MUST be assigned at least one role that contains the desired permission (scope).
+
+If the actor does not specify the organization context, the authorization MUST fail.
+
+It's important to note that organization roles are not considered in authorization. The actor is authorized to perform the action based solely on the organization permission, regardless of their organization role.
```

Comment:
```suggestion
It's important to note that organization roles are not considered in authorization. The actor is authorized to perform the action based solely on the organization permissions, regardless of their organization roles.
```

Reply:
i think here for a specific action, it's mapped to one permission (instead of "permissions"). good with "roles"

Reply:
thinking "permission" would be more appropriate since it's for a specific tenant. i'm good with "permissions"